blocks/library/text/index.js

```diff
@@ -31,13 +31,29 @@ registerBlockType( 'core/text', {
 		content: query( 'p', children() ),
 	},
 
+	transforms: {
+		from: [
+			{
+				type: 'raw',
+				matcher: ( node ) => (
+					node.nodeName === 'P' &&
```

Comment:
We should treat it differently. In a paste context, you're dropping IDs, classes, style attributes, empty elements, etc. while for legacy content you are not.